gap/generic/KnownNilpotent.gi

```diff
@@ -89,10 +93,10 @@ FindHomMethodsGeneric.KnownNilpotent := function(ri,G)
   data := rec( primesfactor := primes{[1..cut]},
                primeskernel := primes{[cut+1..Length(primes)]},
                orderfunc := ri!.order );
-  gens2 := List(gensm,x->RECOG.DecomposeNilpotent(data,x));
-  gensfac := List(gens2,x->StripMemory(x[2]));
-  gensker := List(gens2,x->x[1]);
-  ri!.waytonice := List(gens2,x->x{[3,4]});
+  decompositionData := List(gensm,x->RECOG.DecomposeNilpotent(data,x));
+  gensfac := List(decompositionData,x->StripMemory(x[2]));
+  gensker := List(decompositionData,x->x[1]);
+  ri!.decompositionExponents := List(decompositionData,x->x{[3,4]});
```

Comment:
```suggestion
  ri!.decompositionExponents := List(decompositionData, x -> x{[3,4]});
```

Reply:
Above, we only seem to use `decompositionExponents[i][1]` -- so we could just change this code here to use `x->x[3]` and then above use `decompositionExponents[i]`, no?